### PR TITLE
Update wyre token list

### PR DESCRIPTION
**Pull request by muliswilliam**
Resolves https://github.com/brave/brave-browser/issues/23103

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_wallet/browser/blockchain_registry_unittest.cc [hunk 1]
```diff
@@ -356,7 +356,7 @@ TEST(BlockchainRegistryUnitTest, GetBuyTokens) {
       mojom::OnRampProvider::kWyre, mojom::kPolygonMainnetChainId,
       base::BindLambdaForTesting(
           [&](std::vector<mojom::BlockchainTokenPtr> token_list) {
-            EXPECT_EQ(token_list.size(), 0UL);
+            EXPECT_EQ(token_list.size(), 2.0);
```
Comment (onyb) [inline components/brave_wallet/browser/blockchain_registry_unittest.cc:359]: This works, but let's do `2UL` for consistency. `UL` means unsigned long, by the way.

File: components/brave_wallet/browser/brave_wallet_constants.cc [hunk 2]
```diff
@@ -115,6 +140,9 @@ const std::vector<mojom::BlockchainToken>& GetRampBuyTokens() {
         mojom::CoinType::ETH},
        {"0x0d8775f648430679a709e98d2b0cb6250d2887ef", "Basic Attention Token",
         "bat.png", true, false, "BAT", 18, true, "", "", mojom::kMainnetChainId,
+        mojom::CoinType::ETH},
+       {"EPeUFDgHRxs9xxEPVaL6kfGQvCon7jmAWKVUHuux1Tpz", "Basic Attention Token",
+        "bat.png", true, false, "BAT", 18, true, "", "", mojom::kSolanaMainnet,
```
Comment (onyb) [inline components/brave_wallet/browser/brave_wallet_constants.cc:145]: The `CoinType` should be `SOL`.

**Review by onyb — CHANGES_REQUESTED** [review]

File: components/brave_wallet/browser/blockchain_registry_unittest.cc (hunk 1 above)
Comment (muliswilliam) [reply components/brave_wallet/browser/blockchain_registry_unittest.cc]: @onyb Thank you. Making changes

**Review by onyb — APPROVED** [review]

**Comment by srirambv** [comment]
Verification passed on <html><head></head>
<body>


Brave | 1.41.34 Chromium: 102.0.5005.78 (Official Build) nightly (64-bit)
-- | --
Revision | `df6dbb5a9fd82af3f567198af2eb5fb4876ef99c-refs/branch-heads/5005_59@{#3}`
OS | Windows 11 Version 22H2 (Build 22621.1)


</body>
</html>

- Verified BAT is listed under tokens for Buy on Solana network
- Verified only RAMP is available for BAT under Solana network

https://user-images.githubusercontent.com/17010094/171834793-dbb1ce05-3819-42a0-a465-297804db7f55.mov

**Pull request merged**